Share status

toniengelhardt opened the pull request:
Generated from #27. Closes #20.

- Simplifies status logic
- Tweaks
- Style

paoloricciuti commented:
Ok i've added base 64 encode and decode. I've also fixed a bug in the decode status.

Now the only problem is...i can't push to this branch lol

Would you prefer that i merge this into my fork and we go back to the other PR?

paoloricciuti commented:
Also testing is failing because it can't find webApisData...importing it manually works but than it tryes to import "#components" and still fail. I'm trying to work around this without much success

toniengelhardt commented:
Excellent! 

Yes, let's go back to your PR. Just created the second one bc I also didn't understand how to easily add on top of yours without checking out your fork separately. 

Don't worry, just merge your stuff and I'll take a look at the test.

paoloricciuti commented:
Perfect, pushed on the other PR.

paoloricciuti commented:
@toniengelhardt it seems like it's prerendering index.html...i'm trying to force ssr to see if it fixes it

paoloricciuti commented on pages/index.vue:
```diff
@@ -85,9 +85,14 @@ const totalAPICount = computed(() => webApiList.value.length)
+//   console.log(`DisplayMode changed from '${oldVal}' to '${newVal}'`)
+// })
+
+onMounted(() => {
```
Running this only onMounted will leave us without SSR tho

paoloricciuti replied:
With this https://github.com/toniengelhardt/webapicheck/pull/28#issuecomment-1438582395 we can fix the prerendering issue and get SSR when there's a query param and populating the fields on the client when there's not.

toniengelhardt commented:
Hey @paoloricciuti, I just merged the PR in prod. 

Some notes:
- Linting and tests are working now, but only in the CI, locally I somehow get errors with sitemap.xml (probably related to the nuxt-seo-kit), really strange. I switched the setup from vitest to [nuxt-vitest](https://github.com/danielroe/nuxt-vitest), I think that was the issue.
- I switched the logic back to always `onMounted`. Reasons are the following: it feels off that the `onMounted` hook is in the composable (bad readability) and it will probably cause problems in the future if the logic is not consistent, bc we will always have to deal with 2 cases instead of just one. Since the performance gain is minimal I think this is the better solution and keeps things simple. This also has the advantage that we don't need to distinguish between `btoa` and `Buffer` which will probably also cause issues if the environment is suddenly not node or browser. This way we can just always use `window.btoa`.
- I switched the encoding/decoding from `parseInt` to `BigInt` because there were rounding errors.

For future reference, the changelog is auto-generated, so if you want your commits to be included you need to follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) guidelines (see contribution section in README), like `feat: share button` for example.

All in all, EXCELLENT WORK, I think the feature is really cool! 🙏🏽

paoloricciuti commented:
> Hey @paoloricciuti, I just merged the PR in prod.
> 
> Some notes:
> 
> * Linting and tests are working now, but only in the CI, locally I somehow get errors with sitemap.xml (probably related to the nuxt-seo-kit), really strange. I switched the setup from vitest to [nuxt-vitest](https://github.com/danielroe/nuxt-vitest), I think that was the issue.
> * I switched the logic back to always `onMounted`. Reasons are the following: it feels off that the `onMounted` hook is in the composable (bad readability) and it will probably cause problems in the future if the logic is not consistent, bc we will always have to deal with 2 cases instead of just one. Since the performance gain is minimal I think this is the better solution and keeps things simple. This also has the advantage that we don't need to distinguish between `btoa` and `Buffer` which will probably also cause issues if the environment is suddenly not node or browser. This way we can just always use `window.btoa`.
> * I switched the encoding/decoding from `parseInt` to `BigInt` because there were rounding errors.
> 
> For future reference, the changelog is auto-generated, so if you want your commits to be included you need to follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) guidelines (see contribution section in README), like `feat: share button` for example.
> 
> All in all, EXCELLENT WORK, I think the feature is really cool! 🙏🏽

All right man, sorry for the messy code, first time working with nuxt. If you feel it's better to leave the logic in the onMounted you have all the right to do so 😄 

For the future i'll try to follow the conventional commit rules (it's an habit i have to start following).

Happy to have contributed to your project and thanks for the patience 🙇

toniengelhardt commented:
No worries, I'm actually really impressed what you came up with given you have never worked with Nuxt!

Thanks a lot 🙏🏽 🙏🏽 🙏🏽

paoloricciuti commented:
Thank you for your project 💪
